internal/sync/lifecycle.go

```diff
@@ -115,12 +155,15 @@ func (l *LifecycleOnce) Stop(f func() error) error {
 		return l.stopErr
 	}
 
-	if f == nil {
-		f = func() error { return nil }
+	// Unblock WaitForStart callers (ccounting Stop() without Start())
+	if l.started != nil {
+		close(l.started)
```

Comment:
but we're closing the started channel in both start and stop, so it's possible for us to close the channel twice